src/utils/firebaseSnapshots.tsx

```diff
@@ -0,0 +1,17 @@
+import { getDocs, collection } from "firebase/firestore";
+import db from "../firebase";
+
+const eventsCollectionRef = collection(db, "events");
+
+export const getEventList = async () => {
+  try {
+    const data = await getDocs(eventsCollectionRef);
+    const filteredData = data.docs.map((doc) => ({
```

Comment:
You are going to have to cast this to a MockEvent type as it thinks you are returning a string[] atm.

```tsx
export const getEventList = async () => {
  try {
    const data = await getDocs(eventsCollectionRef);
    const filteredData = data.docs.map((doc) => ({
      ...(doc.data() as MockEvent),
      id: doc.id,
    }));
    return filteredData;
  } catch (err) {
    console.error(err);
  }
};
```

Reply:
You will hit an error that the `MockEvent` has a number as the id when it is now going to be a string You can change the definition to be a string. 
While you are there you can change the name & file to Event rather than MockEvent.

Reply:
You will also have to change the mockdata to use a string that should be all.